micro-deps/micro-deps-spring-config/src/main/java/com/ofg/infrastructure/discovery/ServiceDiscoveryInfrastructureConfiguration.java

```diff
@@ -86,18 +87,34 @@ public ServiceInstance serviceInstance(MicroserviceAddressProvider addressProvid
                 .build();
     }
 
-    @Bean(initMethod = "start", destroyMethod = "close")
+    @Bean(destroyMethod = "close")
     @SuppressWarnings("unchecked")
     public ServiceDiscovery serviceDiscovery(CuratorFramework curatorFramework,
                                              ServiceInstance serviceInstance,
                                              ServiceConfigurationResolver serviceConfigurationResolver) {
         log.info("Registering myself: " + String.valueOf(serviceInstance));
-        return ServiceDiscoveryBuilder
+        final ServiceDiscovery<Void> serviceDiscovery = ServiceDiscoveryBuilder
                 .builder(Void.class)
                 .basePath("/" + serviceConfigurationResolver.getBasePath())
                 .client(curatorFramework)
                 .thisInstance(serviceInstance)
                 .build();
+        asyncRegisterInsideZookeeper(serviceDiscovery);
+        return serviceDiscovery;
+    }
+
+    private void asyncRegisterInsideZookeeper(final ServiceDiscovery<Void> serviceDiscovery) {
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    serviceDiscovery.start();
+                } catch (Exception e) {
+                    log.error("Error during service registration inside Zookeeper");
```

Comment:
After checking called method internals it seems that nearly each error will be handled internally (for example if can't register it will try 29 times with exponential backoff). Adding this can help if internal threads was interrupted so then registration can fail so we want to kill application.